### ISSUE TITLE
test: gb test failed on go devel with matchString has been moved testdeps.TestDeps{}

### ISSUE THREAD

**Issue by zchee** [constabulary/gb#653]
I'm raising an issue before sending a PR.

`gb test` command will fail on go devel at least `+f9d406ebea`.
It seems to cause by Go core `test` source has been moved `matchString` to `testdeps.TestDeps{}`.
The error message is below.

```sh
$ gb test
.
.
# foopackage/testmain
_testmain.go:48: cannot use matchString (type func(string, string) (bool, error)) as type testing.testDeps in argument to testing.MainStart:
        func(string, string) (bool, error) does not implement testing.testDeps (missing MatchString method)
FATAL: command "test" failed: exit status 2
```

---

I don't know what this is correct but will work with patched the below diff. Note that haven't backward compatibility.

```diff
diff --git a/test/gotest.go b/test/gotest.go
index 40ef52c..2aa26a4 100644
--- a/test/gotest.go
+++ b/test/gotest.go
@@ -241,8 +241,8 @@ import (
 {{if not .TestMain}}
 	"os"
 {{end}}
-	"regexp"
 	"testing"
+	"testing/internal/testdeps"
 
 {{if .ImportTest}}
 	{{if .NeedTest}}_test{{else}}_{{end}} {{.Package.ImportPath | printf "%q"}}
@@ -273,24 +273,10 @@ var benchmarks = []testing.InternalBenchmark{
 
 var examples = []testing.InternalExample{
 {{range .Examples}}
-	{Name: "{{.Name}}", F: {{.Package}}.{{.Name}}, Output: {{.Output | printf "%q"}}},
+	{"{{.Name}}", {{.Package}}.{{.Name}}, {{.Output | printf "%q"}}, {{.Unordered}}},
 {{end}}
 }
 
-var matchPat string
-var matchRe *regexp.Regexp
-
-func matchString(pat, str string) (result bool, err error) {
-	if matchRe == nil || matchPat != pat {
-		matchPat = pat
-		matchRe, err = regexp.Compile(matchPat)
-		if err != nil {
-			return
-		}
-	}
-	return matchRe.MatchString(str), nil
-}
-
 {{if .CoverEnabled}}
 
 // Only updated by init functions, so no need for atomicity.
@@ -339,7 +325,7 @@ func main() {
 		CoveredPackages: {{printf "%q" .Covered}},
 	})
 {{end}}
-	m := testing.MainStart(matchString, tests, benchmarks, examples)
+	m := testing.MainStart(testdeps.TestDeps{}, tests, benchmarks, examples)
 {{with .TestMain}}
 	{{.Package}}.{{.Name}}(m)
 {{else}}
```

```sh
go version devel +f9d406ebea Sat Nov 5 01:17:21 2016 +0000 darwin/amd64

GOARCH="amd64"
GOBIN=""
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/zchee/go"
GORACE=""
GOROOT="/usr/local/go"
GOTOOLDIR="/usr/local/go/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/kk/nlghjzs913d70lnngdp8bhsr0000gn/T/go-build360754391=/tmp/go-build -gno-record-gcc-switches -fno-common"
CXX="clang++"
CGO_ENABLED="1"
PKG_CONFIG="pkg-config"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
```

**Comment by davecheney** [comment]
Fixed #657

**Comment by zchee** [comment]
@davecheney Thanks :)
